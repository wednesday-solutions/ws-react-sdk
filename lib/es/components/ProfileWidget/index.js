import{_ as e,d as n,b as t,a as r}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"antd/lib/slider/style";import i from"antd/lib/slider";import"antd/lib/spin/style";import o from"antd/lib/spin";import"antd/lib/message/style";import a from"antd/lib/message";import"antd/lib/button/style";import l from"antd/lib/button";import"antd/lib/modal/style";import d from"antd/lib/modal";import"../_rollup-plugin-inject-process-env-0dfb24f4.js";import c,{useState as m,useCallback as s}from"react";import g from"prop-types";import p,{css as u}from"styled-components";import h from"react-easy-crop";import f from"axios";import v from"../T/index.js";import{c as b}from"../colors-bf7c26a7.js";import{f as w}from"../fonts-56bad7cc.js";import{m as x}from"../media-54d0e07e.js";import{appIntl as y}from"../IntlGlobalProvider/index.js";import{MinusOutlined as E,PlusOutlined as P,LoadingOutlined as _}from"@ant-design/icons";import"react-intl";import"../If/index.js";var j,k,C,z,M,S,I,R,U,H,L,B,F,T,A,O,D,V,q,G,W=function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:"row",t=arguments.length>1&&void 0!==arguments[1]?arguments[1]:"center",r=arguments.length>2&&void 0!==arguments[2]?arguments[2]:"center",i=arguments.length>3&&void 0!==arguments[3]?arguments[3]:"center",o=arguments.length>4&&void 0!==arguments[4]?arguments[4]:0,a=arguments.length>5&&void 0!==arguments[5]?arguments[5]:1,l=arguments.length>6&&void 0!==arguments[6]?arguments[6]:0;return u(j||(j=e(["\n  ","\n  flex-direction: ",";\n  justify-content: ",";\n  align-items: ",";\n  align-content: ",";\n  flex-basis: ",";\n  flex-grow: ",";\n  flex-shrink: ",";\n"])),"row"===n?Z():J(),n,t,r,i,o,a,l)},Z=()=>u(k||(k=e(["\n  display: flex;\n  flex: 1;\n  flex-direction: row;\n"]))),J=()=>u(C||(C=e(["\n  display: flex;\n  flex: 1;\n  flex-direction: column;\n"]))),K=n=>u(A||(A=e(["\n    border-radius: ",";\n  "])),n+"".concat("string"==typeof n?";":"px")),N={height:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:4;return u(S||(S=e(["\n  height: ","rem;\n"])),n)},zIndex:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:1;u(q||(q=e(["\n    z-index: ",";\n  "])),n)},textEllipsis:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:"200px";return u(G||(G=e(["\n  white-space: nowrap;\n  overflow: hidden;\n  width: ",";\n  text-overflow: ellipsis;\n"])),n)},margin:{top:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:0;return u(I||(I=e(["\n    margin-top: ","rem;\n  "])),n)},bottom:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:0;return u(R||(R=e(["\n    margin-bottom: ","rem;\n  "])),n)},right:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:0;return u(H||(H=e(["\n    margin-right: ","rem;\n  "])),n)},left:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:0;return u(U||(U=e(["\n    margin-left: ","rem;\n  "])),n)},vertical:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:0;return u(L||(L=e(["\n  margin-top: ","rem;\n  margin-bottom: ","rem;\n"])),n,n)},horizontal:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:0;return u(B||(B=e(["\n  margin-left: ","rem;\n  margin-right: ","rem;\n"])),n,n)}},borderWithRadius:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:1,t=arguments.length>1&&void 0!==arguments[1]?arguments[1]:"solid",r=arguments.length>2&&void 0!==arguments[2]?arguments[2]:"#ccc",i=arguments.length>3&&void 0!==arguments[3]?arguments[3]:0;return u(O||(O=e(["\n    border: ","px "," ",";\n    ","\n  "])),n,t,r,K(i))},borderRadius:K,borderRadiusBottom:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:0;return u(F||(F=e(["\n  border-bottom-left-radius: ","px;\n  border-bottom-right-radius: ","px;\n"])),n,n)},borderRadiusTop:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:0;return u(T||(T=e(["\n  border-top-left-radius: ","px;\n  border-top-right-radius: ","px;\n"])),n,n)},primaryBackgroundColor:()=>u(V||(V=e(["\n    background-color: ",";\n  "])),b.accent),flexConfig:{row:Z,column:J,rowCenter:()=>u(z||(z=e(["\n  ",";\n"])),W("row","center","center","center")),unequalColumns:()=>u(M||(M=e(["\n  ",";\n"])),W("column","","","",0,0,0))},boxShadow:function(){var n=arguments.length>0&&void 0!==arguments[0]?arguments[0]:0,t=arguments.length>1&&void 0!==arguments[1]?arguments[1]:0,r=arguments.length>2&&void 0!==arguments[2]?arguments[2]:0,i=arguments.length>3&&void 0!==arguments[3]?arguments[3]:0,o=arguments.length>4&&void 0!==arguments[4]?arguments[4]:"#ccc";return u(D||(D=e(["\n    box-shadow: ","px ","px ","px ","px ",";\n  "])),n,t,r,i,o)}},Q=e=>new Promise(((n,t)=>{var r=new Image;r.addEventListener("load",(()=>n(r))),r.addEventListener("error",(e=>t(e))),r.setAttribute("crossOrigin","anonymous"),r.src=e}));function X(e){return e*Math.PI/180}function Y(){return(Y=n((function*(e,n){var t=arguments.length>2&&void 0!==arguments[2]?arguments[2]:0,r=yield Q(e),i=document.createElement("canvas"),o=i.getContext("2d"),a=Math.max(r.width,r.height),l=a/2*Math.sqrt(2)*2;i.width=l,i.height=l,o.translate(l/2,l/2),o.rotate(X(t)),o.translate(-l/2,-l/2),o.drawImage(r,l/2-.5*r.width,l/2-.5*r.height);var d=o.getImageData(0,0,l,l);return i.width=n.width,i.height=n.height,o.putImageData(d,Math.round(0-l/2+.5*r.width-n.x),Math.round(0-l/2+.5*r.height-n.y)),new Promise((e=>{i.toBlob((n=>{e(URL.createObjectURL(n))}),"image/jpeg")}))}))).apply(this,arguments)}var $,ee,ne,te,re,ie,oe,ae,le,de,ce,me,se,ge,pe,ue,he,fe=p.div($||($=e(["\n  width: 6rem;\n  position: relative;\n  border-radius: 10px;\n  align-self: center;\n  margin-bottom: 20px;\n"]))),ve=p.div(ee||(ee=e(["\n  width: 100%;\n  ",";\n  align-items: center;\n  justify-content: center;\n"])),N.flexConfig.row()),be=p.div(ne||(ne=e(["\n  width: 5.625rem;\n  height: 5.625rem;\n  position: relative;\n"]))),we=p.img(te||(te=e(["\n  width: 100%;\n  height: 100%;\n  border-radius: 50%;\n"]))),xe=p.div(re||(re=e(["\n  width: 100%;\n  height: 100%;\n  border-radius: 50%;\n  padding: 4px;\n  background-color: ",";\n  ",";\n  align-items: center;\n  justify-content: center;\n"])),b.primary,N.flexConfig.row()),ye=p(v)(ie||(ie=e(["\n  margin-bottom: 0;\n  font-weight: 500;\n  font-size: 4.5rem;\n  line-height: 4.5rem;\n  color: ",";\n  display: inline-block;\n"])),b.white),Ee=p.div(oe||(oe=e(["\n  width: 0.5rem;\n  height: 0.5rem;\n  display: flex;\n  float: right;\n  cursor: pointer;\n"]))),Pe=p.div(ae||(ae=e(["\n  height: 1.5rem;\n  width: 1.5rem;\n  border-radius: 50%;\n  display: flex;\n  align-items: center;\n  background: ",";\n  cursor: pointer;\n"])),b.whiteSmoke),_e=p.img(le||(le=e(["\n  width: 1rem;\n  height: 1rem;\n  margin-left: 0.25rem;\n"]))),je=p.input(de||(de=e(["\n  width: 0.1px;\n  height: 0.1px;\n  opacity: 0;\n  overflow: hidden;\n  position: absolute;\n  z-index: -1;\n"]))),ke=p.div(ce||(ce=e(["\n  width: 18rem;\n  height: 18rem;\n  position: relative;\n"]))),Ce=p.div(me||(me=e(["\n  ","\n  width: 100%;\n  height: 100%;\n  ",";\n  align-items: center;\n  justify-content: center;\n  border-radius: 50%;\n  opacity: 1;\n"])),(e=>e.bgi&&"background-image: url(".concat(e.bgi,");")),N.flexConfig.row()),ze=p(d)(se||(se=e(["\n  .ant-modal-content {\n    border-radius: 8px;\n  }\n  .ant-modal-body {\n    padding-bottom: 0;\n  }\n  .ant-modal-footer {\n    border: 0;\n    margin-top: 1rem;\n    padding: 1.5rem;\n  }\n"]))),Me=p(l)(ge||(ge=e(["\n  color: ",";\n  ",";\n  font-stretch: normal;\n  line-height: 1;\n  letter-spacing: normal;\n"])),b.pawlyBlue,w.weights.bold()),Se=p(l)(pe||(pe=e(["\n  ",";\n  font-stretch: normal;\n  line-height: 1;\n  letter-spacing: normal;\n  border-radius: 5px;\n"])),w.weights.bold()),Ie=p.div(ue||(ue=e(["\n  background: ",";\n  padding: 1rem;\n  margin: 0.1rem;\n  ","\n"])),b.whiteSmoke,x.largeMobile.max("\n    padding: 0.6rem;\n  ")),Re=p.div(he||(he=e(["\n  position: relative;\n  padding: 0px 2rem;\n  .anticon {\n    position: absolute;\n    top: -2px;\n    color: ",";\n    font-size: 16px;\n    line-height: 1;\n  }\n  .anticon:first-child {\n    left: 0;\n  }\n  .anticon:last-child {\n    right: 0;\n  }\n"])),b.pawlyBlue),Ue=c.createElement(_,{style:{fontSize:50},spin:!0});function He(e){var{user:l,updateProfilePicture:d,loading:g,onProfileImgSet:p,disabled:u}=e,b=t(e,["user","updateProfilePicture","loading","onProfileImgSet","disabled"]),[w,x]=m(null),[_,j]=m(!1),[k,C]=m({x:0,y:0}),[z,M]=m(1),[S,I]=m(null),[R,U]=m(null),[H,L]=m(""),B=function(){var e=n((function*(e,n){var t={headers:{"Content-Type":H.type}};try{var r=yield f.put(e,n,t);if(200===r.status){var i=r.config.url.substr(0,r.config.url.indexOf("?"));d(i)}}catch(e){a.error(y().formatMessage({id:"profile_picture_updated_fail"}))}}));return function(n,t){return e.apply(this,arguments)}}(),F=s(((e,n)=>{I(n)}),[]),T=s(function(){var e=n((function*(e){try{var n=yield function(e,n){return Y.apply(this,arguments)}(w,S),t=yield fetch(n).then((e=>e.blob()));yield O(e,t),U(n),p(n),D()}catch(e){D()}}));return function(n){return e.apply(this,arguments)}}(),[w,S]),A=function(){var e=n((function*(e){if(e.target.files&&e.target.files.length>0){var n=e.target.files[0],t=yield function(e){return new Promise((n=>{var t=new FileReader;t.addEventListener("load",(()=>n(t.result)),!1),t.readAsDataURL(e)}))}(n);x(t),L(n);if(n.size>65536)return void a.error(y().formatMessage({id:"profile_picture_wrong_size"}));j(!0);["image/png","image/jpeg","image/jpg"].includes(n.type)||(a.error(y().formatMessage({id:"profile_picture_wrong_format"})),D())}}));return function(n){return e.apply(this,arguments)}}(),O=(e,n)=>new Promise(((e,t)=>b.getPreSignedUri({contentType:H.type.replace("/","_")}).then((r=>{try{var{data:{preSignedUri:i}}=r,o=new File([n],i.key,{type:H.type});B(i.uri,o).then((()=>{e(i.uri)}))}catch(e){a.error(y().formatMessage({id:"profile_picture_updated_fail"})),t(e)}})))),D=()=>{j(!1)};var V,q=()=>{T(H.name)};return c.createElement(fe,r({},b,{"data-testid":"profile-widget"}),c.createElement(ve,null,c.createElement(be,null,!u&&c.createElement(Ee,null,c.createElement("label",{htmlFor:"profilePic"},c.createElement(Pe,null,c.createElement(_e,{src:"data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='16' height='16.002' viewBox='0 0 16 16.002'%3e %3cdefs%3e %3cstyle%3e .cls-1%7bfill:none%7d.cls-4%7bstroke:none%7d %3c/style%3e %3c/defs%3e %3cg id='ic_edit' transform='rotate(-54.97 14.414 7.125)'%3e %3cpath id='Rectangle_957' d='M0 0H16V16H0z' class='cls-1' transform='rotate(54.97 14.407 7.134)'/%3e %3cg id='Group_1629' fill='%233164c6' transform='translate(1.721 .866)'%3e %3cg id='Union_12'%3e %3cpath d='M15.002 2.593H2.642L.951 1.546 2.642.5h12.36v2.093z' class='cls-4' transform='translate(0 .003)'/%3e %3cpath d='M14.502 2.093V1H2.784l-.883.546.883.547h11.718m1 1H2.5L0 1.546 2.5 0h13.002v3.093z' transform='translate(0 .003)'/%3e %3c/g%3e %3cg id='Rectangle_952' stroke='%233164c6' transform='translate(16.001)'%3e %3cpath d='M0 0H1V3.092H0z' class='cls-4'/%3e %3crect height='2.092' x='.5' y='.5' class='cls-1'/%3e %3c/g%3e %3c/g%3e %3c/g%3e%3c/svg%3e"}))),c.createElement(je,{"data-cy":"profilePic",type:"file",name:"profilePic",id:"profilePic",accept:"image/jpeg, image/jpg, image/png",onChange:A,onClick:e=>e.target.value=null})),R?c.createElement(Ce,{bgi:R},g&&c.createElement(o,{indicator:Ue})):null!=l&&l.photoUrl?c.createElement(we,{src:null==l?void 0:l.photoUrl}):c.createElement("span",{"data-testid":"user-name-initial"},c.createElement(xe,null,c.createElement(ye,{text:null==l||null===(V=l.name)||void 0===V?void 0:V.charAt(0)}))))),c.createElement(ze,{onCancel:D,visible:_,title:null,width:"23rem",centered:!0,closable:!1,onOk:q,footer:[c.createElement(Me,{key:"back",type:"text",onClick:D},c.createElement(v,{id:"cancel"})),c.createElement(Se,{key:"submit",type:"primary",loading:g,onClick:q},c.createElement(v,{id:"upload_new"}))]},c.createElement(Ie,null,c.createElement(ke,null,c.createElement(h,{image:w,crop:k,zoom:z,aspect:1,onCropChange:C,onCropComplete:F,onZoomChange:M,cropSize:{width:256,height:256},cropShape:"round"}))),c.createElement(Re,null,c.createElement(E,null),c.createElement(i,{min:1,max:3,step:.1,tipFormatter:null,onChange:function(e){M(e)}}),c.createElement(P,null))))}He.propTypes={onSetActiveForm:g.func,activeForm:g.number,user:g.object,updateProfilePicture:g.func,loading:g.bool,onProfileImgSet:g.func,getPreSignedUri:g.func,round:g.func,disabled:g.boolean};export default He;
