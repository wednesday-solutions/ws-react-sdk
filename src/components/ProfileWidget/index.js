/**
 *
 * ProfileWidget
 *
 */

import React, { useCallback, useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { message, Modal, Spin, Button, Slider } from 'antd';
import Cropper from 'react-easy-crop';
import axios from 'axios';
import T from '@components/T';
import { colors, styles, fonts, media } from '@themes';
import { getCroppedImg } from '@utils/canvasUtils';
import { appIntl } from '@components/IntlGlobalProvider';
import iconEdit from '@images/ic-edit.svg';
import { MinusOutlined, PlusOutlined, LoadingOutlined } from '@ant-design/icons';
const Container = styled.div`
  width: 6rem;
  position: relative;
  border-radius: 10px;
  align-self: center;
  margin-bottom: 20px;
`;

const ProfileImageContainer = styled.div`
  width: 100%;
  ${styles.flexConfig.row()};
  align-items: center;
  justify-content: center;
`;

const ProfileImageWrapper = styled.div`
  width: 5.625rem;
  height: 5.625rem;
  position: relative;
`;

const ProfileImage = styled.img`
  width: 100%;
  height: 100%;
  border-radius: 50%;
`;

const ProfileImageTextWrapper = styled.div`
  width: 100%;
  height: 100%;
  border-radius: 50%;
  padding: 4px;
  background-color: ${colors.primary};
  ${styles.flexConfig.row()};
  align-items: center;
  justify-content: center;
`;

const ProfileImageText = styled(T)`
  margin-bottom: 0;
  font-weight: 500;
  font-size: 4.5rem;
  line-height: 4.5rem;
  color: ${colors.white};
  display: inline-block;
`;

const ProfileImageEditWrapper = styled.div`
  width: 0.5rem;
  height: 0.5rem;
  display: flex;
  float: right;
  cursor: pointer;
`;
const ProfileIconContainer = styled.div`
  height: 1.5rem;
  width: 1.5rem;
  border-radius: 50%;
  display: flex;
  align-items: center;
  background: ${colors.whiteSmoke};
  cursor: pointer;
`;
const ProfileImageEditIcon = styled.img`
  width: 1rem;
  height: 1rem;
  margin-left: 0.25rem;
`;

const FileElement = styled.input`
  width: 0.1px;
  height: 0.1px;
  opacity: 0;
  overflow: hidden;
  position: absolute;
  z-index: -1;
`;

const CropWrapper = styled.div`
  width: 18rem;
  height: 18rem;
  position: relative;
`;
const CustomOverlay = styled.div`
  ${props => props.bgi && `background-image: url(${props.bgi});`}
  width: 100%;
  height: 100%;
  ${styles.flexConfig.row()};
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  opacity: 1;
`;
const StyledModal = styled(Modal)`
  .ant-modal-content {
    border-radius: 8px;
  }
  .ant-modal-body {
    padding-bottom: 0;
  }
  .ant-modal-footer {
    border: 0;
    margin-top: 1rem;
    padding: 1.5rem;
  }
`;
const CancelButton = styled(Button)`
  color: ${colors.pawlyBlue};
  ${fonts.weights.bold()};
  font-stretch: normal;
  line-height: 1;
  letter-spacing: normal;
`;
const SubmitButton = styled(Button)`
  ${fonts.weights.bold()};
  font-stretch: normal;
  line-height: 1;
  letter-spacing: normal;
  border-radius: 5px;
`;
const CropDiv = styled.div`
  background: ${colors.whiteSmoke};
  padding: 1rem;
  margin: 0.1rem;
  ${media.largeMobile.max(`
    padding: 0.6rem;
  `)}
`;
const SliderDiv = styled.div`
  position: relative;
  padding: 0px 2rem;
  .anticon {
    position: absolute;
    top: -2px;
    color: ${colors.pawlyBlue};
    font-size: 16px;
    line-height: 1;
  }
  .anticon:first-child {
    left: 0;
  }
  .anticon:last-child {
    right: 0;
  }
`;
const antIcon = <LoadingOutlined style={{ fontSize: 50 }} spin />;

function ProfileWidget({ user, updateProfilePicture, loading, onProfileImgSet, disabled, ...props }) {
  const [imageSrc, setImageSrc] = useState(null);
  const [showCropModal, setShowCropModal] = useState(false);
  const [crop, setCrop] = useState({ x: 0, y: 0 });
  const [zoom, setZoom] = useState(1);
  const [croppedAreaPixels, setCroppedAreaPixels] = useState(null);
  const [croppedImage, setCroppedImage] = useState(null);
  const [imgData, setImgData] = useState('');

  const uploadProfilePicture = async (preSignedPutUrl, data) => {
    const config = {
      headers: {
        'Content-Type': imgData.type
      }
    };
    try {
      const response = await axios.put(preSignedPutUrl, data, config);
      if (response.status === 200) {
        const fileUrl = response.config.url.substr(0, response.config.url.indexOf('?'));
        updateProfilePicture(fileUrl);
      }
    } catch (e) {
      message.error(appIntl().formatMessage({ id: 'profile_picture_updated_fail' }));
    }
  };

  const onCropComplete = useCallback((croppedArea, croppedAreaPixels) => {
    setCroppedAreaPixels(croppedAreaPixels);
  }, []);

  const showCroppedImage = useCallback(
    async fileName => {
      try {
        const ci = await getCroppedImg(imageSrc, croppedAreaPixels);
        const blob = await fetch(ci).then(r => r.blob());
        await handleUpload(fileName, blob);
        setCroppedImage(ci);
        onProfileImgSet(ci);
        onCloseCropModal();
      } catch {
        onCloseCropModal();
      }
    },
    [imageSrc, croppedAreaPixels]
  );
  const onFileChange = async e => {
    if (e.target.files && e.target.files.length > 0) {
      const file = e.target.files[0];
      const imageDataUrl = await readFile(file);
      setImageSrc(imageDataUrl);
      setImgData(file);
      const allowedImgSize = 256 * 256;
      if (file.size > allowedImgSize) {
        message.error(appIntl().formatMessage({ id: 'profile_picture_wrong_size' }));
        return;
      }
      setShowCropModal(true);
      const allowedImgTypes = ['image/png', 'image/jpeg', 'image/jpg'];
      if (!allowedImgTypes.includes(file.type)) {
        message.error(appIntl().formatMessage({ id: 'profile_picture_wrong_format' }));
        onCloseCropModal();
      }
    }
  };
  const handleUpload = (fileName, blob) =>
    new Promise((resolve, reject) => {
      return props.getPreSignedUri({ contentType: imgData.type.replace('/', '_') }).then(response => {
        try {
          const {
            data: { preSignedUri }
          } = response;
          const file = new File([blob], preSignedUri.key, {
            type: imgData.type
          });
          uploadProfilePicture(preSignedUri.uri, file).then(() => {
            resolve(preSignedUri.uri);
          });
        } catch (err) {
          message.error(appIntl().formatMessage({ id: 'profile_picture_updated_fail' }));
          reject(err);
        }
      });
    });

  const onCloseCropModal = () => {
    setShowCropModal(false);
  };

  function readFile(file) {
    return new Promise(resolve => {
      const reader = new FileReader();
      reader.addEventListener('load', () => resolve(reader.result), false);
      reader.readAsDataURL(file);
    });
  }

  const renderProfileImage = () => {
    if (croppedImage) {
      return <CustomOverlay bgi={croppedImage}>{loading && <Spin indicator={antIcon} />}</CustomOverlay>;
    }
    if (user?.photoUrl) {
      return <ProfileImage src={user?.photoUrl} />;
    }
    return (
      <span data-testid="user-name-initial">
        <ProfileImageTextWrapper>
          <ProfileImageText text={user?.name?.charAt(0)} />
        </ProfileImageTextWrapper>
      </span>
    );
  };
  const handleNewImage = () => {
    showCroppedImage(imgData.name);
  };
  function onChangeSlider(value) {
    setZoom(value);
  }
  return (
    <Container {...props} data-testid="profile-widget">
      <ProfileImageContainer>
        <ProfileImageWrapper>
          {!disabled && (
            <ProfileImageEditWrapper>
              <label htmlFor="profilePic">
                <ProfileIconContainer>
                  <ProfileImageEditIcon src={iconEdit} />
                </ProfileIconContainer>
              </label>

              <FileElement
                data-cy="profilePic"
                type="file"
                name="profilePic"
                id="profilePic"
                accept="image/jpeg, image/jpg, image/png"
                onChange={onFileChange}
                onClick={event => (event.target.value = null)}
              />
            </ProfileImageEditWrapper>
          )}
          {renderProfileImage()}
        </ProfileImageWrapper>
      </ProfileImageContainer>

      <StyledModal
        onCancel={onCloseCropModal}
        visible={showCropModal}
        title={null}
        width="23rem"
        centered
        closable={false}
        onOk={handleNewImage}
        footer={[
          <CancelButton key="back" type="text" onClick={onCloseCropModal}>
            <T id="cancel" />
          </CancelButton>,
          <SubmitButton key="submit" type="primary" loading={loading} onClick={handleNewImage}>
            <T id="upload_new" />
          </SubmitButton>
        ]}
      >
        <CropDiv>
          <CropWrapper>
            <Cropper
              image={imageSrc}
              crop={crop}
              zoom={zoom}
              aspect={1}
              onCropChange={setCrop}
              onCropComplete={onCropComplete}
              onZoomChange={setZoom}
              cropSize={{ width: 256, height: 256 }}
              cropShape={'round'}
            />
          </CropWrapper>
        </CropDiv>
        <SliderDiv>
          <MinusOutlined />
          <Slider min={1} max={3} step={0.1} tipFormatter={null} onChange={onChangeSlider} />
          <PlusOutlined />
        </SliderDiv>
      </StyledModal>
    </Container>
  );
}

ProfileWidget.propTypes = {
  onSetActiveForm: PropTypes.func,
  activeForm: PropTypes.number,
  user: PropTypes.object,
  updateProfilePicture: PropTypes.func,
  loading: PropTypes.bool,
  onProfileImgSet: PropTypes.func,
  getPreSignedUri: PropTypes.func,
  round: PropTypes.func,
  disabled: PropTypes.boolean
};

export default ProfileWidget;
