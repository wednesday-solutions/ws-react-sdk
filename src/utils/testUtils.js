import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
