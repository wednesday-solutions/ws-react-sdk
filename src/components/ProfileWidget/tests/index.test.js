/**
 *
 * Tests for ProfileWidget
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import ProfileWidget from '../index';

describe('<ProfileWidget />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<ProfileWidget />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 ProfileWidget component', () => {
    const { getAllByTestId } = renderWithIntl(<ProfileWidget />);
    expect(getAllByTestId('profile-widget').length).toBe(1);
  });
});
