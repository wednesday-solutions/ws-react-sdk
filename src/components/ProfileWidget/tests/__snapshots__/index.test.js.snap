// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileWidget /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-gsTCUz jhMctP"
      data-testid="profile-widget"
    >
      <div
        class="sc-dlfnbm bezHed"
      >
        <div
          class="sc-hKgILt epGQjk"
        >
          <div
            class="sc-iBPRYJ drIJeq"
          >
            <label
              for="profilePic"
            >
              <div
                class="sc-fubCfw jkfXYs"
              >
                <img
                  class="sc-pFZIQ iUSsxT"
                  src="IMAGE_MOCK"
                />
              </div>
            </label>
            <input
              accept="image/jpeg, image/jpg, image/png"
              class="sc-jrAGrp gmVUeL"
              data-cy="profilePic"
              id="profilePic"
              name="profilePic"
              type="file"
            />
          </div>
          <span
            data-testid="user-name-initial"
          >
            <div
              class="sc-jSgupP jQoqQN"
            >
              <p
                class="sc-bdfBwQ bZojWc sc-gKsewC hdfsZu"
                data-testid="t"
              />
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
